Content.Client/GameObjects/Components/Clothing/ClothingComponent.cs

```diff
@@ -56,6 +57,54 @@ public override void Initialize()
         {
             base.Initialize();
             ClothingEquippedPrefix = ClothingEquippedPrefix;
+
+            // Get parent RSIs to fill in missing sprites
+            if (!Owner.TryGetComponent(out SharedItemComponent? item))
+            {
+                return;
+            }
+
+            var rsiPath = item.RsiPath;
+            if (rsiPath == null)
+            {
+                return;
+            }
+
+            var rsi = IoCManager.Resolve<IResourceCache>().GetResource<RSIResource>(SharedSpriteComponent.TextureRoot / rsiPath).RSI;
+
+            var prototypeManager = IoCManager.Resolve<IPrototypeManager>();
+
+            FillInParentSprites(prototypeManager, Owner.Prototype?.Parent, rsi);
+        }
+
+        private void FillInParentSprites(IPrototypeManager prototypeManager, string? parentPrototypeName, RSI rsi)
+        {
+            if (parentPrototypeName == null)
+            {
+                return;
+            }
+
+            var prototype = prototypeManager.Index<EntityPrototype>(parentPrototypeName);
+            if (!prototype.TryGetComponent("Clothing", out ClothingComponent? parentClothes) || parentClothes.RsiPath == null)
+            {
+                return;
+            }
+
+            var parentRsi = IoCManager.Resolve<IResourceCache>().GetResource<RSIResource>(SharedSpriteComponent.TextureRoot / parentClothes.RsiPath).RSI;
+            var enumerator = parentRsi.GetEnumerator();
+
+            while (enumerator.MoveNext())
+            {
+                var arg = enumerator.Current;
+
+                if (!rsi.TryGetState(arg.StateId, out var dummy))
+                {
+                    rsi.AddState(arg);
+                }
+            }
+            enumerator.Dispose();
+
+            FillInParentSprites(prototypeManager, prototype.Parent, rsi);
```

Comment:
is the problem here that the new jumpsuits are missing textures and this is a way to implement fallbacks? Because this seems like a terrible terrible workaround.